### ISSUE TITLE
Extract features about EPIC100

### ISSUE THREAD

**Issue by EdenGabriel** [facebookresearch/omnivore#35]
Hi,
Thanks for your work, i have an question about extracting features. Could you please tell me how to extract features by using [Omnivore](https://github.com/facebookresearch/omnivore/blob/main/omnivore) on epic100.

**Comment by rohitgirdhar** [comment]
Hi,
Thanks for your interest in this work. Please see [here](https://github.com/facebookresearch/omnivore/tree/main/omnivore#run-inference) for inference instructions (including on egocentric videos)

**Comment by EdenGabriel** [comment]
Thanks for your reply, i have implemented inference process in [here](https://github.com/facebookresearch/omnivore/tree/main/omnivore#run-inference)
![image](https://user-images.githubusercontent.com/39000840/203375473-6af9d70e-3617-400b-a83c-2a38f504e4e4.png)
but i also have some questions, if i just want features about epic, whether i just need to remove classification heads? That is, the extracted feature dim is 1024 for 2s video.

![image](https://user-images.githubusercontent.com/39000840/203376043-b81569a7-7020-42cf-bb29-0dd9f1daa70a.png)
In addition, if i also want to get the logits of _verb_ and _noun_, not action, how should i do it?  
Looking forward to your replay!

**Comment by rohitgirdhar** [comment]
Yes, if you just want the last layer features (before the classification layer), remove the `heads` from this model.
So in our model we don't separately compute logits for verbs and nouns, and only train with predictions over the joint distribution of verbs x nouns ("actions"). We obtain the final probabilities over verbs and nouns separately by marginalizing over the other (i.e. after normalizing the logits with softmax, we sum over all verbs corresponding to each noun to get the probability of that noun).

**Comment by EdenGabriel** [comment]
Thank you vvvvvvvvvery much for your patient replay. I got it.
Also, could you please give me some advice about extracting features? _i.e._ extract a feature vector every 0.5 second?
Looking forward to your replay. Thanks.

**Comment by rohitgirdhar** [comment]
That would depend on the downstream application. For eg the Ego4D project (https://ego4d-data.org/docs/data/features/) released Omnivore features extracted every 1 second which seems good enough for a wide range of tasks.

**Comment by EdenGabriel** [comment]
ok, i got it.
Thank you for your excellent work and replay again.

**Comment by Vincent6896** [comment]
> Thanks for your reply, i have implemented inference process in [here](https://github.com/facebookresearch/omnivore/tree/main/omnivore#run-inference) ![image](https://user-images.githubusercontent.com/39000840/203375473-6af9d70e-3617-400b-a83c-2a38f504e4e4.png) but i also have some questions, if i just want features about epic, whether i just need to remove classification heads? That is, the extracted feature dim is 1024 for 2s video.
> 
> ![image](https://user-images.githubusercontent.com/39000840/203376043-b81569a7-7020-42cf-bb29-0dd9f1daa70a.png) In addition, if i also want to get the logits of _verb_ and _noun_, not action, how should i do it? Looking forward to your replay!

Hi May I ask how do you turn off the heads from the model? I turn the load_heads to false, but when model.eval() will return a None which is incorrect. Could you please give some advice on that? Thank you very much.

**Comment by rohitgirdhar** [comment]
Hi @Vincent6896 you can load the whole model and use the approach from https://discuss.pytorch.org/t/how-to-delete-layer-in-pretrained-model/17648 to remove layers.

**Comment by zeyun-zhong** [comment]
Thanks for the great work and the released models.

For future readers: We extracted the omnivore features on EPIC100 (for each timestamp, the last 32 frames corresponding to 1.067 seconds are fed into the omnivore model). Feel free to download the features [here](https://github.com/zeyun-zhong/AFFT#download-data).
